apps/e2e/helpers/cr-sqlite.ts

```diff
@@ -1,55 +1,69 @@
 import type { DB } from "@vlcn.io/crsqlite-wasm";
-import {
-	Customer,
-	PlacedSupplierOrder,
-	PlacedSupplierOrderLine,
-	Supplier,
-	PossibleSupplierOrderLine,
-	ReconciliationOrderLine,
-	ReconciliationOrder,
-	DBCustomerOrderLine
-} from "./types";
+
+import { Customer, PlacedSupplierOrder, PlacedSupplierOrderLine, Supplier, CustomerOrderLine, SupplierOrderLine } from "./types";
 import { BookData } from "@librocco/shared";
 
 // #region books
+import { Page } from "@playwright/test";
+
+import type {} from "./types";
+
+// #region books
+
+// Extend the window object with the db
+declare global {
+	interface Window {
+		_db: DB;
+		db_ready: boolean;
+
+		books: Record<string, any>;
+		customers: Record<string, any>;
+		note: Record<string, any>;
+		reconciliation: Record<string, any>;
+		suppliers: Record<string, any>;
+		warehouse: Record<string, any>;
+	}
+}
+
+/**
+ * Returns the database handle from the db interface registered in the window object
+ * of the app. We can run `hadle.evaluate` to run queries/mutations against the database.
+ * @example
+ * ```ts
+ * const dbHandle = await getDbHandle(page);
+ *
+ * // Use the handle to create a warehouse in the db
+ * await dbHandle.evaluate(async (db) => {
+ *   await db.warehouse("foo-wh").create()
+ * })
+ * ```
+ */
+export function getDbHandle(page: Page) {
+	return page.evaluateHandle(async () => {
+		const w = window as { [key: string]: any };
+
+		// Wait for the db to become initialised
+		await new Promise<void>((res) => {
+			if (w["db_ready"]) {
+				return res();
+			} else {
+				// Creating a separate function, as we want to run the listener only once and then remove it
+				const finalise = () => {
+					window.removeEventListener("db_ready", finalise), res();
```

Comment:
I was about to comment that this `finalize` function does nothing, since it only removes the event listener that just invoked it. But then I noticed the `,res()` promise resolution.

I think it would be more readable in its own line, with a semicolon.

```suggestion
					window.removeEventListener("db_ready", finalise);
					res();
```

...but I fear I might be missing something.

Reply:
No, you're right - this was a one liner turned into a block (resulting in this retarded hybrid)